test/Microsoft.AspNetCore.DataProtection.AzureKeyVault.Test/AzureKeyVaultXmlEncryptorTests.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Security.Cryptography;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using Microsoft.Azure.KeyVault.Models;
+using Microsoft.Azure.KeyVault.WebKey;
+using Microsoft.Extensions.DependencyInjection;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNetCore.DataProtection.AzureKeyVault.Test
+{
+    public class AzureKeyVaultXmlEncryptorTests
+    {
+        [Fact]
+        public void UsesKeyVaultToEncryptKey()
+        {
+            var mock = new Mock<IKeyVaultWrappingClient>();
+            mock.Setup(client => client.WrapKeyAsync("key", JsonWebKeyEncryptionAlgorithm.RSAOAEP, It.IsAny<byte[]>()))
+                .Returns<string, string, byte[]>((_, __, data) => Task.FromResult(new KeyOperationResult("KeyId", data.Reverse().ToArray())));
+
+            var encryptor = new AzureKeyVaultXmlEncryptor(mock.Object, "key", new MockNumberGenerator());
+            var result = encryptor.Encrypt(new XElement("Element"));
+
+            var encryptedElement = result.EncryptedElement;
+            var value = encryptedElement.Element("value");
+
+            mock.VerifyAll();
+            Assert.NotNull(result);
+            Assert.NotNull(value);
+            Assert.Equal(typeof(AzureKeyVaultXmlDecryptor), result.DecryptorType);
```

Comment:
Nit: `Assert.IsType<AzureKeyVaultXmlDecryptor>(result.DecryptorType)`.

Reply:
This would be wrong, `IsType` checks type of an object, I'm checking `Type` instance equality.